oracle/config/aggregate_market.go

```diff
@@ -0,0 +1,186 @@
+package config
+
+import (
+	"fmt"
+
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+)
+
+// AggregateMarketConfig represents the market configurations for how currency pairs will
+// be resolved to a final price. Each currency pair can have a list of convertable markets
+// that will be used to convert the price of the currency pair to a common currency pair.
+//
+// Price aggregation broadly follows the following steps:
+//  1. Fetch prices for each currency pair from the price feeds.
+//  2. Calculate the provider-weighted median price for each currency pair.
+//  3. Convert the price of each currency pair to a common currency pair using the
+//     AggregatedFeeds field. If there are multiple convertable feeds for a given currency
+//     pair, calculate a median price for each final currency pair, weighted by the
+//     weights of the convertable feeds.
+type AggregateMarketConfig struct {
```

Comment:
We can add an additional `Weights` field here that defines the relative weight we should associate with each provider.

Reply:
Add linear issue to track?